src/main/java/graphql/util/LinkedHashMapFactory.java

```diff
@@ -0,0 +1,344 @@
+package graphql.util;
+
+import graphql.Internal;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Factory class for creating LinkedHashMap instances with insertion order preservation.
+ * Use this instead of Map.of() to ensure consistent serialization order.
+ * <p>
+ * This class provides static factory methods similar to Map.of() but returns mutable LinkedHashMap
+ * instances that preserve insertion order, which is important for consistent serialization.
+ */
+@Internal
+public final class LinkedHashMapFactory {
```

Comment:
This is verbose. Originally I was only going to have one method (the last one) with variable length parameters, but saw the original Map.java had all of these extra methods for performance.

I don't think the difference is massive for our purposes, but I thought I'd follow the original